### DISCUSSION TITLE
Specific example for google auth?

### DISCUSSION THREAD

**Discussion by ppbrown** [netbox-community/netbox#9078] [Q&A] [answered]
In the docs here and there, are references such as
static/docs/administration/authentication/

> REMOTE_AUTH_BACKEND = 'social_core.backends.google.GoogleOAuth2'
> 
> Most remote authentication backends require some additional configuration through settings prefixed with SOCIAL_AUTH_
> 

But nowhere can I find specific examples of what this "additional configuration" should *be*, for google oauth.
The links on that page reference external stuff such as
https://github.com/python-social-auth

but nowhere that I can find, is the netbox-specific adjustments required.

Could someone post an example, please?

I'm using netbox 3.1.9

**Comment by ppbrown** [comment, marked as answer]
Not nice that I had to sort of guess where to put this.
Seems like it works, if I just stuff
SOCIAL_AUTH_GOOGLE_OAUTH2_KEY = 'xxxx'
SOCIAL_AUTH_GOOGLE_OAUTH2_SECRET = 'xxxx'

right after 
REMOTE_AUTH_BACKEND = 'social_core.backends.google.GoogleOAuth2'
in configuration.py

Side note, that isnt directly part of my original question, but still quite important:
It wont let you just make up your own internal DNS FQDNs. Google oauth2 insists on a recognizable top level domain like ".com" ...
even if it cant resolve  privateserver.yourdomain.com

Otherwise it kvetches about "server does not meet google oauth2 security standards". which makes no sense to me, but...

**Comment by ianb-mp** [comment]
I was wondering the same thing. In addition to the settings indicated above (I put them in `extra.py`) I also had to specify these settings when configuring the oauth2 app at https://console.cloud.google.com/ :
- Authorized JavaScript origins: `http://<myhost>`  (yes, http is acceptable however as @ppbrown said, must use a TLD like `.com`)
- Authorized redirect URIs: `http://<myhost>/oauth/complete/google-oauth2/`
